app/command_autocomplete.go

```diff
@@ -4,109 +4,189 @@
 package app
 
 import (
+	"net/url"
+	"sort"
 	"strings"
 
+	"github.com/mattermost/mattermost-server/v5/mlog"
 	"github.com/mattermost/mattermost-server/v5/model"
 )
 
 // GetSuggestions returns suggestions for user input.
-func (a *App) GetSuggestions(commands []*model.AutocompleteData, userInput, roleID string) []model.AutocompleteSuggestion {
+func (a *App) GetSuggestions(commands []*model.Command, userInput, roleID string) []model.AutocompleteSuggestion {
+	sort.Slice(commands, func(i, j int) bool {
```

Comment:
1/5 cache (or normalize) the lowercase representations to avoid repetitive lowercasing. A separate ticket is ok.

Reply:
Normalizing automatically will not do, because if content is uppercase we want to output suggestions in uppercase(e.g. `/jira assign MM-12320 Lev`). We will still autocomplete even if user types in lowercase.